pkg/generated-assets/sanitize.go

```diff
@@ -1,16 +1,43 @@
 package generated_assets
 
 import (
+	"bytes"
+
 	sigyaml "sigs.k8s.io/yaml"
 )
 
 // Sanitize reorders fields in a YAML file in a canonical order, so they can be compared easily with `diff`.
+// Preserves comments at the beginning of the file, but all other comments will be lost.
 func Sanitize(src []byte) ([]byte, error) {
+
+	comments, err := initialComments(src)
+	if err != nil {
+		return nil, err
+	}
+
 	var obj interface{}
 	sigyaml.Unmarshal(src, &obj)
-	bytes, err := sigyaml.Marshal(obj)
+	sanitized, err := sigyaml.Marshal(obj)
 	if err != nil {
 		return nil, err
 	}
-	return bytes, nil
+	return bytes.Join([][]byte{comments, sanitized}, []byte{'\n'}), nil
+}
+
+// initialComments returns the comments at the beginning of the file.
+func initialComments(src []byte) ([]byte, error) {
+	buf := bytes.NewBuffer(src)
+	comments := bytes.NewBuffer(nil)
+	for {
+		line, err := buf.ReadBytes('\n')
+		if err != nil {
+			return nil, err
+		}
+		if line[0] != '#' {
+			return comments.Bytes(), nil
+		}
+		comments.Write(line)
+	}
+
+	return nil, nil
```

Comment:
`go vet` is unhappy because this line is unreachable. I'd suggest:
```
- return nil, nil
- // We can never be here.
```
I'd also appreciate a comment explaining your attitude toward asset files with comments only (w/o actual content). Something short and simple like this would work for me:
```
// We don't expect asset yaml files containing only comments. initialComments() would fail in such cases intentionally.
```

Reply:
Good point about files containing only comments. I added a separate error for that.
(+ there should be an unit test for this...)